src/common_types/src/datum.rs

```diff
@@ -765,6 +776,11 @@ impl Datum {
         }
     }
 
+    fn valid_is_utf8(s: &str) -> Result<()> {
+        from_utf8(s.as_bytes()).context(InvalidStringEncoding { msg: s })?;
```

Comment:
IMO this function should return `bool`, not a result.

Reply:
ok